sdk/servicebus/service-bus/.scripts/buildBrowserTestResources.js

```diff
@@ -0,0 +1,352 @@
+          getNamespace(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]),
+          queueName,
+          parameters,
+          function(error) {
+            if (error) throw error.message;
+          }
+        );
+      });
+  }
+
+  async function recreateTopic(topicName, parameters) {
+    await msRestNodeAuth
+      .loginWithServicePrincipalSecret(
+        env[EnvVarKeys.AAD_CLIENT_ID],
+        env[EnvVarKeys.AAD_CLIENT_SECRET],
+        env[EnvVarKeys.AAD_TENANT_ID]
+      )
+      .then(async (creds) => {
+        const client = await new ServiceBusManagementClient(
+          creds,
+          env[EnvVarKeys.AZURE_SUBSCRIPTION_ID]
+        );
+        await client.topics.deleteMethod(
+          env[EnvVarKeys.RESOURCE_GROUP],
+          getNamespace(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]),
+          topicName,
+          function(error) {
+            if (error) throw error.message;
+          }
+        );
+        await client.topics.createOrUpdate(
+          env[EnvVarKeys.RESOURCE_GROUP],
+          getNamespace(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]),
+          topicName,
+          parameters,
+          function(error) {
+            if (error) throw error.message;
+          }
+        );
+      });
+  }
+
+  async function recreateSubscription(topicName, subscriptionName, parameters) {
+    await msRestNodeAuth
+      .loginWithServicePrincipalSecret(
+        env[EnvVarKeys.AAD_CLIENT_ID],
+        env[EnvVarKeys.AAD_CLIENT_SECRET],
+        env[EnvVarKeys.AAD_TENANT_ID]
+      )
+      .then(async (creds) => {
+        const client = await new ServiceBusManagementClient(
+          creds,
+          env[EnvVarKeys.AZURE_SUBSCRIPTION_ID]
+        );
+        /*
+        Unlike Queues/Topics, there is no need to delete the subscription because
+        `recreateTopic` is called before `recreateSubscription` which would
+        delete the topic and the subscriptions before creating a new topic.
+      */
+        await client.subscriptions.createOrUpdate(
+          env[EnvVarKeys.RESOURCE_GROUP],
+          getNamespace(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]),
+          topicName,
+          subscriptionName,
+          parameters,
+          function(error) {
+            if (error) throw error.message;
+          }
+        );
+      });
+  }
+}
+
+/**
+ * Utility function to get namespace string from given connection string
+ * @param serviceBusConnectionString
+ */
+function getNamespace(serviceBusConnectionString) {
+  return (serviceBusConnectionString.match("Endpoint=sb://(.*).servicebus.windows.net") || "")[1];
+}
+
+function throwMissingEnvironmentVariablesError(envVars) {
+  envVars.forEach(function(key) {
+    const name = key.valueOf();
+    if (!getEnvVarValue(name)) {
+      throw new Error(`Define ${name} in your environment before running integration tests.`);
+    }
+  });
+}
+
+function getEnvVarValue(name, forBrowser) {
+  if (forBrowser) {
+    name = name + "_BROWSER";
```

Comment:
Since we are dealing only with browser, do we need `getEnvVarValue` to exist at all?

Reply:
It's just reusing the envVarUtils logic.

Ideally I wanted to reuse that envVarUtils code directly but ts-node had an issue and we wanted to do duplicate.
I think the duplication is already leads to being pretty error prone. Would avoid optimizing, changing this much as that would take more time to track diff versions of same env var accessing. Issues can be easily understood, fixed if things remain consistent i.e., in case anything wrong is identified with the overall accessing of env variables.

Reply:
The overall accessing of env variables that you had refactored last month is pretty stable. It has remained stable for months before the refactoring as well. So, I wouldn't worry too much about that.

This script is named build**Browser**TestResources. It is meant explicitly for setting up resources for the browser. Having the code that looks like it supports both browser and non-browser modes will only cause confusion to the next person who might land on this code for any reason.

Your original intent to re-use the code from `envVarUtils.ts` file was noble indeed. But we soon realized that it was not worth the effort and having a simple stand-alone javascript file was easier to work with, especially whey we need to be run as part of a npm script.

My intent here is mainly simplicity.

Now, if the intent for this script was to work for both node and browser, then we can update the entire script to take an argument "isBrowser". But that is not the case. Please have this script to be as simple as possible.

Reply:
Yes, I was trying to explain that it might complicate things in that the default values are being defined and used differently, and it may be confusing to make changes to these later.
But yes, overall anyone will need to understand all the references and usage of browser related changes and we do indicate the value we are passing for 'forBrowser' is true - but i'm okay with making this file more browser specific